diff -ruN '--exclude=mod.patch' migration/v15/model/application.go migration/v16/model/application.go
--- migration/v15/model/application.go	2025-01-21 08:38:57.215536535 -0600
+++ migration/v16/model/application.go	2025-04-10 09:15:06.091516415 -0500
@@ -174,19 +174,16 @@
 	Model
 	UUID       *string `gorm:"uniqueIndex"`
 	Name       string  `gorm:"uniqueIndex:tagA;not null"`
-	Username   string
-	CategoryID uint `gorm:"uniqueIndex:tagA;index;not null"`
+	CategoryID uint    `gorm:"uniqueIndex:tagA;index;not null"`
 	Category   TagCategory
 }
 
 type TagCategory struct {
 	Model
-	UUID     *string `gorm:"uniqueIndex"`
-	Name     string  `gorm:"index;unique;not null"`
-	Username string
-	Rank     uint
-	Color    string
-	Tags     []Tag `gorm:"foreignKey:CategoryID;constraint:OnDelete:CASCADE"`
+	UUID  *string `gorm:"uniqueIndex"`
+	Name  string  `gorm:"index;unique;not null"`
+	Color string
+	Tags  []Tag `gorm:"foreignKey:CategoryID;constraint:OnDelete:CASCADE"`
 }
 
 type Ticket struct {
